src/lazyload-image.directive.ts

```diff
@@ -47,8 +48,8 @@ export class LazyLoadImageDirective implements OnChanges, AfterContentInit, OnDe
   }
 
   ngAfterContentInit() {
-    // Disable lazy load image in server side
-    if (isPlatformServer(this.platformId)) {
+    // Don't do anything if SSR and the user isn't a bot
+    if (isPlatformServer(this.platformId) && !this.hooks.isBot(getNavigator())) {
```

Comment:
Why not pass platformId into `hooks.isBot` and allow people to decide if they want images or not?

The problem right now is that Universal doesn't render the page with images and you have to wait for JS to get downloaded before they even start loading. Above the fold responses images only appear until a couple of seconds in so you get a janky experience

Reply:
Actually... my mistake. I realise I can just always return true in the isBot function :)

Reply:
Hi @intellix

> Why not pass platformId into hooks.isBot and allow people to decide if they want images or not?

That is a great idea! I should have thought of that. PR is welcome.

> I realise I can just always return true in the isBot function

If you always return true, you will get the [ssrPreset](https://github.com/tjoskar/ng-lazyload-image/blob/master/src/ssr-preset/preset.ts) which is probably not what you want.

https://github.com/tjoskar/ng-lazyload-image/blob/a81a0ab45ffdf7fbf7d2abae061d6ec32308b7f9/src/hooks-factory.ts#L10-L14

> This assumes we only want bots to have images for SEO but we also want a great loading experience for users

I don't follow. If you always want the images to be loaded right away, then I don't see the purpose of lazyloading. What am I missing?

Reply:
We should create a new issue for this.

Reply:
done :) https://github.com/tjoskar/ng-lazyload-image/issues/414

Reply:
Added a reproduction in the above issue. I don't think this commit works at all :)